From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sebastian Birunt <sebastian.birunt@arm.com>
Date: Thu, 5 Sep 2024 11:47:52 +0200
Subject: fix: remove `dupbuild` flag

Newer `ninja-build` does not have `dupbuild`
warning flag anymore:
https://github.com/ninja-build/ninja/pull/2356

diff --git a/blueprint_impl.bash b/blueprint_impl.bash
index 6f5abba..d950d96 100644
--- a/blueprint_impl.bash
+++ b/blueprint_impl.bash
@@ -30,15 +30,21 @@ BUILDDIR="${BUILDDIR}/.minibootstrap" build_go minibp github.com/google/blueprin

 BUILDDIR="${BUILDDIR}/.minibootstrap" build_go bpglob github.com/google/blueprint/bootstrap/bpglob

+NINJA_ARGS=()
+# Newer `ninja` does not have `dupbuild` warning flag
+if ! test "${NINJA} -w list | grep -q '^  dupbuild'"; then
+  NINJA_ARGS+=( "-w" "dupbuild=err" )
+fi
+
 # Build the bootstrap build.ninja
-"${NINJA}" -w dupbuild=err -f "${BUILDDIR}/.minibootstrap/build.ninja"
+"${NINJA}" ${NINJA_ARGS[@]} -f "${BUILDDIR}/.minibootstrap/build.ninja"

 # Build the primary builder and the main build.ninja
-"${NINJA}" -w dupbuild=err -f "${BUILDDIR}/.bootstrap/build.ninja"
+"${NINJA}" ${NINJA_ARGS[@]} -f "${BUILDDIR}/.bootstrap/build.ninja"

 # SKIP_NINJA can be used by wrappers that wish to run ninja themselves.
 if [ -z "$SKIP_NINJA" ]; then
-    "${NINJA}" -w dupbuild=err -f "${BUILDDIR}/build.ninja" "$@"
+    "${NINJA}" ${NINJA_ARGS[@]} -f "${BUILDDIR}/build.ninja" "$@"
 else
     exit 0
 fi
--
2.43.2
